docs/developer-docs/daos/sns/launching/launch-summary-1proposal.mdx

```diff
@@ -90,6 +90,12 @@ If successful, at the end of stage, the following has changed:
   </tr>
 </table>
 
+:::warning
+Adding NNS Root as co-controller is necessary because NNS Root needs to be able to make sure that no one else has the ability to change the canisters during the swap. This is automatically done in a later step which sets NNS Root and SNS Root as the only controllers of the dapp canisters. The prohibition of changing the canisters during the swap is necessary to ensure that the swap participants get what they expect.
+
+This means that it is not easy to upgrade an SNS-controlled dapp canister during the swap. However, since the SNS-controlled dapp canisters are co-owned by NNS root during a swap, it is still possible via an NNS proposal. This is intended to be used on an emergency basis and should be avoided if possible. Once the swap is over, SNS Root takes sole control of the dapp canisters and futher upgrades are performed through SNS proposals.
```

Comment:
```suggestion
This means that it is not easy to upgrade an SNS-controlled dapp canister during the swap. However, since the SNS-controlled dapp canisters are co-owned by NNS root during a swap, it is still possible via an NNS proposal in emergency situations. This is intended to be used on an emergency basis and should be avoided if possible. Once the swap is over, SNS Root takes sole control of the dapp canisters and futher upgrades are performed through SNS proposals.
```